### PR TITLE
virtio: Fix the problem of incorrect setting of virtio queue address under label kasan

### PR DESCRIPTION
**Pull request by W-M-R**
…under label kasan

*Note: Please adhere to [Contributing Guidelines](https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md).*

## Summary

virtio: Fix the problem of incorrect setting of virtio queue address under label kasan

## Impact

*Update this section, where applicable, on how change affects users,
 build process, hardware, documentation, security, compatibility, etc.*

## Testing

*Update this section with details on how did you verify the change,
 what Host was used for build (OS, CPU, compiler, ..), what Target was
 used for verification (arch, board:config, ..), etc. Providing build
 and runtime logs from before and after change is highly appreciated.*

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



This PR appears **incomplete** and does **not** fully meet the NuttX requirements.  While it provides a summary of the change, it lacks crucial information in the Impact and Testing sections.

Here's a breakdown:

* **Summary:** Adequate, though a more detailed explanation of *how* the virtio queue address was incorrectly set and *how* the fix corrects it would be beneficial.

* **Impact:**  Completely empty. This section needs to be filled out.  At minimum, consider:
    * **Impact on hardware:** This fix likely affects virtio devices, so specify which architectures and boards are impacted.
    * **Impact on compatibility:** Does this fix introduce any backward compatibility issues with older virtio drivers or configurations?
    * **Impact on security:**  Incorrect memory addressing could have security implications.  Even if the answer is NO, explicitly state it and provide a brief justification.

* **Testing:**  Empty. This is a critical deficiency.  Provide details on:
    * **Build Host:** OS, CPU architecture, compiler version.
    * **Target:**  Architecture (e.g., QEMU, a specific RISC-V board), board configuration.
    * **Logs:**  Include logs demonstrating the issue *before* the change and showing the correct behavior *after* the change. These logs should ideally show the virtio queue addresses.

**To make this PR acceptable, the Impact and Testing sections must be completed with specific details.** Simply stating "NO" for all impact categories is insufficient; a brief explanation is necessary, especially if the impact is indeed minimal.  Without testing logs and a description of the test setup, reviewers cannot effectively evaluate the correctness of the proposed fix.

File: drivers/virtio/virtio-mmio.c [hunk 1]
```diff
@@ -343,16 +344,22 @@ static int virtio_mmio_config_virtqueue(FAR struct metal_io_region *io,
       metal_io_write32(io, VIRTIO_MMIO_QUEUE_NUM, vq->vq_nentries);
 
       addr = (uint64_t)(uintptr_t)vq->vq_ring.desc;
-      metal_io_write32(io, VIRTIO_MMIO_QUEUE_DESC_LOW, addr);
-      metal_io_write32(io, VIRTIO_MMIO_QUEUE_DESC_HIGH, addr >> 32);
+      metal_io_write32(io, VIRTIO_MMIO_QUEUE_DESC_LOW,
+                       (uint64_t)kasan_reset_tag((void *)addr));
```
Comment (CV-Bowen) [inline drivers/virtio/virtio-mmio.c:348]: FAR

File: drivers/virtio/virtio-mmio.c [hunk 2]
```diff
@@ -343,16 +344,22 @@ static int virtio_mmio_config_virtqueue(FAR struct metal_io_region *io,
       metal_io_write32(io, VIRTIO_MMIO_QUEUE_NUM, vq->vq_nentries);
 
       addr = (uint64_t)(uintptr_t)vq->vq_ring.desc;
```
Comment (anchao) [inline drivers/virtio/virtio-mmio.c:346]:
```suggestion
      addr = (uint64_t)kasan_reset_tag((FAR void *)vq->vq_ring.desc);
```

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by lupyuen — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
let's ignore ci temp download fail:
```
====================================================================================
Configuration/Tool: esp32-devkitc/nxdiag
2024-12-05 07:34:54
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Building NuttX...
fatal: could not read Username for 'https://github.com/': No such device or address
make[3]: *** [Makefile:83: espressif_prepare] Error 128
make[3]: Target 'context' not remade because of errors.
make[2]: *** [Makefile:55: /github/workspace/sources/apps/system/nxdiag_context] Error 2
make[2]: Target 'context_all' not remade because of errors.
make[1]: *** [Makefile:185: context] Error 2
make: *** [tools/Unix.mk:457: /github/workspace/sources/apps/.context] Error 2
make: Target 'all' not remade because of errors.
/github/workspace/sources/nuttx/tools/testbuild.sh: line 385: /github/workspace/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
  Normalize esp32-devkitc/nxdiag
```

**Comment by lupyuen** [comment]
Sorry @W-M-R `rv-virt/virt_nsh` fails to build due to a pointer cast error (from nuttx-dashboard.org)
https://github.com/NuttX/nuttx/actions/runs/12188258880/job/34000958824#step:7:485
```text
Configuration/Tool: rv-virt/virt_nsh
virtio/virtio-mmio.c: In function 'virtio_mmio_config_virtqueue':
Error: virtio/virtio-mmio.c:346:14: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  346 |       addr = (uint64_t)kasan_reset_tag((FAR void *)vq->vq_ring.desc);
      |              ^
Error: virtio/virtio-mmio.c:350:14: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  350 |       addr = (uint64_t)kasan_reset_tag((FAR void *)vq->vq_ring.avail);
      |              ^
Error: virtio/virtio-mmio.c:354:14: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  354 |       addr = (uint64_t)kasan_reset_tag((FAR void *)vq->vq_ring.used);
      |              ^
cc1: all warnings being treated as errors
make[1]: *** [Makefile:109: virtio-mmio.o] Error 1
make[1]: Target 'libdrivers.a' not remade because of errors.
make: *** [tools/LibTargets.mk:107: drivers/libdrivers.a] Error 2
```

**Pull request merged**